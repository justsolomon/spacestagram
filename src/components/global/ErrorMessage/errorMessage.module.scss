@import "styles/variables";
@import "styles/mixins";

.error-message {
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;

  &__description {
    margin: 1rem 0;
    font-size: $font-size--md;
    line-height: $font-size--xl;

    @media #{$media-mobile-sm} {
      font-size: $font-size--base;
    }
  }

  &__button {
    @include default-button();

    margin: 0;
    background-color: $brand-color--primary;
    color: $white;
    border-radius: 20px;
    padding: 0 1rem;
    font-weight: $font-weight--medium;
    font-size: $font-size--md;

    &:hover {
      background-color: $brand-color--secondary;
    }

    &:focus-visible {
      outline: none;
      box-shadow: $shadow-outline;
    }

    &:active {
      background-color: $brand-color--tertiary;
    }

    svg {
      height: 1.5em;
      width: 1.5em;
      margin: 0;
      margin-right: 0.3rem;
      color: $white;
    }
  }
}
