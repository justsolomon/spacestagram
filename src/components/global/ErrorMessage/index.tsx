import styles from "./errorMessage.module.scss";
import { ReactComponent as ErrorIcon } from "assets/vectors/error.svg";
import { ReactComponent as RetryIcon } from "assets/vectors/retry.svg";

interface IErrorMessage {
  /** Error message */
  error: string;

  /** Method used to re-run the request */
  retryRequest: () => void;
}

function ErrorMessage({ retryRequest, error }: IErrorMessage) {
  return (
    <div className={styles["error-message"]}>
      <ErrorIcon />
      <p
        className={styles["error-message__description"]}
        data-testid="error-message__description"
      >
        {error === "Network Error"
          ? "Looks like you lost your connection. Please check it and try again"
          : error}
      </p>
      <button
        className={styles["error-message__button"]}
        data-testid="error-message__button"
        onClick={retryRequest}
      >
        <RetryIcon />
        <span>Try again</span>
      </button>
    </div>
  );
}

export default ErrorMessage;
