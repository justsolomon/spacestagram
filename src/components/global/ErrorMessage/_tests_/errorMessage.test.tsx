import { render, screen } from "@testing-library/react";
import ErrorMessage from "..";

describe("<ErrorMessage />", () => {
  it("should render the error message component correctly", () => {
    const { asFragment } = render(
      <ErrorMessage error="An error occurred" retryRequest={() => {}} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
  it("should render a default message for network errors", () => {
    render(<ErrorMessage error="Network Error" retryRequest={() => {}} />);

    const description = screen.queryByTestId("error-message__description");

    expect(description).toHaveTextContent(
      "Looks like you lost your connection. Please check it and try again",
    );
  });
});
