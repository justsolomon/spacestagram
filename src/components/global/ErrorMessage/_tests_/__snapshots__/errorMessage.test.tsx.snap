// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render the error message component correctly 1`] = `
<DocumentFragment>
  <div
    class="error-message"
  >
    <svg>
      error.svg
    </svg>
    <p
      class="error-message__description"
      data-testid="error-message__description"
    >
      An error occurred
    </p>
    <button
      class="error-message__button"
      data-testid="error-message__button"
    >
      <svg>
        retry.svg
      </svg>
      <span>
        Try again
      </span>
    </button>
  </div>
</DocumentFragment>
`;
